coalib/bearlib/spacing/SpacingHelper.py

```diff
@@ -86,11 +86,28 @@ def replace_spaces_with_tabs(self, line: str):
         :param line: The string with spaces to replace.
         :return:     The converted string.
         """
+        def previous_whitespace():
+            # Find the previous real character, and remaining chars to fill tab
```

Comment:
I'm flexible.  I dont know of any significant problems with nested funcs in Python 3 (they do have problems in Python 2).

But I dislike class level funcs unless it uses the `cls` for obvious reasons.  Is a module level private func ok?